src/udev/udev-builtin-net_id.c

```diff
@@ -591,39 +587,50 @@ static int names_platform(sd_device *dev, const char *prefix, bool test) {
         if (r < 0)
                 return log_device_debug_errno(dev, r, "sd_device_get_syspath() failed: %m");
 
-        /* syspath is too short, to have a valid ACPI instance */
-        if (strlen(syspath) < STRLEN(PLATFORM_TEST) + 1)
-                return log_device_debug_errno(dev, SYNTHETIC_ERRNO(EINVAL),
-                                              "Syspath \"%s\" is too short for a valid ACPI instance.",
-                                              syspath);
+        syspath = path_startswith(syspath, "/sys/devices/platform/");
+        if (!syspath)
+                return -EINVAL;
 
-        /* Vendor ID can be either PNP ID (3 chars A-Z) or ACPI ID (4 chars A-Z and numerals) */
-        if (syspath[STRLEN(PLATFORM_TEST)] == ':') {
-                pattern = PLATFORM_PATTERN4;
-                validchars = UPPERCASE_LETTERS DIGITS;
-        } else {
-                pattern = PLATFORM_PATTERN3;
-                validchars = UPPERCASE_LETTERS;
-        }
+        r = path_find_first_component(&syspath, /* accept_dot_dot = */ false, &p);
+        if (r < 0)
+                return r;
 
         /* Platform devices are named after ACPI table match, and instance id
-         * eg. "/sys/devices/platform/HISI00C2:00");
+         * eg. "/sys/devices/platform/HISI00C2:00"
          * The Vendor (3 or 4 char), followed by hexadecimal model number : instance id. */
-
-        DISABLE_WARNING_FORMAT_NONLITERAL;
-        r = sscanf(syspath, pattern, vendor, &model, &instance, &ethid);
-        REENABLE_WARNING;
-        log_device_debug(dev, "Parsing platform device information from syspath \"%s\": %s",
-                         syspath, r == 4 ? "success" : "failure");
-        if (r != 4)
-                return -EINVAL;
+        if (p[7] == ':') {
```

Comment:
uh, this is potentially a bad memory access, in case the the string is not 8 bytes long at least. need to check the length first

Reply:
Ouch. Fix is waiting in #28749.